src/helpers/throw_runtime_error/index.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+import { Errors } from 'Constants';
+const passableArgs:string = 'passableArgs';
+
+function errorBuilder(functionName, errorMessage) {
+    return `[Passable]: Failed to execute '${functionName}': ${errorMessage}`;
+}
+
+function throwRuntimeError(type: string, ...args: Array<string>) {
+    switch (type) {
+        case Errors.INVALID_FORM_NAME:
+            throw new TypeError(errorBuilder('Passable constructor', `Unexpected ${args[0]}, expected string.`));
+        case Errors.INVALID_FORM_NAME:
+            throw new Error(errorBuilder('Enforce', `${args[0]} - invalid ${args[1]} value.`));
+        case Errors.EXPECT_TYPE_FAILURE:
+            const val:string = Array.isArray(args[1]) ? JSON.stringify(args[1]) : args[1];
+            throw new TypeError(errorBuilder(args[0], `expected ${val} to be a ${args[2]}.`));
+        case Errors.PASSABLE_ARGS_NO_ARGS:
+            throw new TypeError(errorBuilder(passableArgs, 'At least 1 argument required, but only 0 present.'));
```

Comment:
Traditionally, wrong number of arguments is not considered a TypeError - but an Error.
Alternatively, you can create an ArgumentError (that inherits Error), but it is uncommon

Reply:
@omrilotan I'm not familiar with that tradition.

Also, if I take `setTimeout()` for example:

In the browser I get:
`Uncaught TypeError: Failed to execute 'setTimeout' on 'Window': 1 argument required, but only 0 present.`

In Node I get a direct reference to the type of argument expected :
`TypeError: "callback" argument must be a function`

I also get TypeErrors in other browser only methods:
`localStorage.setItem() - Uncaught TypeError: Failed to execute 'setItem' on 'Storage': 2 arguments required, but only 0 present.`

`atob() - Failed to execute 'atob' on 'Window': 1 argument required, but only 0 present.`

`btoa() - Uncaught TypeError: Failed to execute 'btoa' on 'Window': 1 argument required, but only 0 present.`